src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php

```diff
@@ -0,0 +1,453 @@
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/collections' . '/(?P<id>[\/\w-]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_font_collection' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Gets a font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_font_collection( $request ) {
+		$id         = $request->get_param( 'id' );
+		$collection = WP_Font_Library::get_font_collection( $id );
+		// If the collection doesn't exist returns a 404.
+		if ( is_wp_error( $collection ) ) {
+			$collection->add_data( array( 'status' => 404 ) );
+			return $collection;
+		}
+		$collection_with_data = $collection->get_data();
+		// If there was an error getting the collection data, return the error.
+		if ( is_wp_error( $collection_with_data ) ) {
+			$collection_with_data->add_data( array( 'status' => 500 ) );
+			return $collection_with_data;
+		}
+		return new WP_REST_Response( $collection_with_data );
+	}
+
+	/**
+	 * Gets the font collections available.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_font_collections() {
+		$collections = array();
+		foreach ( WP_Font_Library::get_font_collections() as $collection ) {
+			$collections[] = $collection->get_config();
+		}
+
+		return new WP_REST_Response( $collections, 200 );
+	}
+
+	/**
+	 * Returns validation errors in font families data for installation.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[] $font_families Font families to install.
+	 * @param array   $files         Files to install.
+	 * @return array $error_messages Array of error messages.
+	 */
+	private function get_validation_errors( $font_families, $files ) {
+		$error_messages = array();
+
+		if ( ! is_array( $font_families ) ) {
+			$error_messages[] = __( 'fontFamilies should be an array of font families.' );
+			return $error_messages;
+		}
+
+		// Checks if there is at least one font family.
+		if ( count( $font_families ) < 1 ) {
+			$error_messages[] = __( 'fontFamilies should have at least one font family definition.' );
+			return $error_messages;
+		}
+
+		for ( $family_index = 0; $family_index < count( $font_families ); $family_index++ ) {
+			$font_family = $font_families[ $family_index ];
+
+			if (
+				! isset( $font_family['slug'] ) ||
+				! isset( $font_family['name'] ) ||
+				! isset( $font_family['fontFamily'] )
+			) {
+				$error_messages[] = sprintf(
+					// translators: 1: font family index.
```

Comment:
```suggestion
					/* translators: %s: font family index. */
```
Translator comments are in the form of:
```php
/* translators: <token || number>: Description<comma || period> */
```

🔢 Applies elsewhere in the PR.

See [the information for Plugins, the best reference I could find 😅](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#variables). This is the same practice used in WordPress Core.